### PR TITLE
Remove Email/get when cleaning mailbox

### PR DESCRIPTION
**Pull request by hoangdat**

### CODE REVIEW

File: lib/features/thread/data/network/thread_api.dart [hunk 1]
```diff
@@ -303,4 +309,91 @@ class ThreadAPI {
...
+
+    final destroyed = setEmailResponse?.destroyed
+      ?.map((id) => EmailId(id));
+    
+    final queriedCount = queryResponse?.ids.length ?? 0;
+    final destroyedCount = destroyed?.length ?? 0;
+    
+    if (destroyedCount < queriedCount) {
+      final notFoundErrorCounter = _countingNotFoundInNotDestroy(setEmailResponse);
+      final isSuccess = destroyedCount + notFoundErrorCounter >= queriedCount;
```
Comment (tddang-linagora) [inline lib/features/thread/data/network/thread_api.dart:373]: Would `destroyedCount + notFoundErrorCounter > queriedCount` ever happens?

File: lib/features/thread/data/network/thread_isolate_worker.dart [hunk 2]
```diff
@@ -71,91 +62,53 @@ class ThreadIsolateWorker {
...
-
+      bool batchResult = true;
       var hasEmails = true;
-      Email? lastEmail;
 
       while (hasEmails) {
-        final emailsResponse = await args.threadAPI.getAllEmail(
-          args.session,
+        final emptyMailboxResponse = await args.threadAPI.deleteEmailsBaseOnQuery(
+          args.session,
```
Comment (tddang-linagora) [inline lib/features/thread/data/network/thread_isolate_worker.dart:70]: Whitespace

File: lib/features/thread/data/network/thread_api.dart (hunk 1 above)
Comment (hoangdat) [reply lib/features/thread/data/network/thread_api.dart:373]: yeah, a safe condition

**Review by tddang-linagora — APPROVED** [review]

**Comment by hoangdat** [comment]
apply empty mailbox with paging #3388

**Pull request closed without merging**